tempesta_fw/http_parser.c

```diff
@@ -282,6 +282,48 @@ do {									\
 	}								\
 } while (0)
 
+/* This macros is like __FSM_MOVE_nofixup_n(), but for interior states. */
+#define __FSM_I_MOVE_nofixup_n(to, n)					\
+do {									\
+	parser->_i_st = to;						\
+	p += n;								\
+	if (unlikely(__data_off(p) >= len)) {				\
+		__fsm_const_state = to; /* start from state @to next time */\
+		__FSM_EXIT(TFW_POSTPONE);				\
+	}								\
+	goto to;							\
+} while (0)
+
+/*
+ * Macros below is like __FSM_MOVE_nf(), but for interior
+ * states and for creation chunk's @ptr not from beginning @data
+ * but from some offset.
+ */
+#define __FSM_I_MOVE_offset_fn(to, n, off_ptr)				\
```

Comment:
The name of the macro doesn't really follow the naming scheme that already exists in the source file. There are macros that end with `_nf()`, and those mean that the move forward is for `n` bytes, and the work is on the field given as an argument.

With that in mind, this `_fn()` ending looks confusing.